### DISCUSSION TITLE
How to update wordpress docker wp-config.php database details on the fly

### DISCUSSION THREAD

**Discussion by shankar-ccm** [docker-library/wordpress#838] [General]
I am deploying custom wordpress docker image to ecs fargate services for each deployment it is asking for database setup details.But i want to update these values while creating docker image and deploy so that it automatically gets added and will not ask for database details to setup.

My database details are available in aws secretmanger which I am fecthing in entrypoint.sh file but not able to append values in wp-config.php.

Even my docker container is only has wp-config-docker.php and wp-config-sample.php but not wp-config.php.

Please help me to overcome this issue.
Thank you!!

**Comment by yosifkit** [comment]
The provided image [entrypoint](https://github.com/docker-library/wordpress/blob/6fdd0eb67cefbad247fd51723ee6f785e2598d5a/docker-entrypoint.sh) will copy `wp-config-docker.php` over to the current directory (should be `/var/ww/html`) as `wp-config.php` if `wp-config.php` does not exist yet.  The `wp-config-docker.php` will read from the environment variables as described in the [readme](https://github.com/docker-library/docs/tree/388783024e937bd87ee2f786fa308dc2ff123099/wordpress#how-to-use-this-image) displayed on the Docker Hub image page (https://hub.docker.com/_/wordpress/).  Many of them even support a [`_FILE` suffix](https://github.com/docker-library/docs/tree/388783024e937bd87ee2f786fa308dc2ff123099/wordpress#docker-secrets) if it should be read from a file instead.

**Reply by shankar-ccm** [reply]
Thank you for the response!!

What I did was while deploying, I am creating ECS task definition which allows to add environment varibales so I added DB values/ARN's to that and then container started picking these values from there and I am no longer having db setup reptition issue for each deployment.

This approach works very well with custom wordpress docker deployment to AWS ECS Fargate Services.

But we do have some more wordpress sites which we are deploying to EC2 instances manually by copying wp-content every time on server.

Now I want to automate this with wordpress docker but I couldn't find the way to add environment variables automatically to container like in AWS ECS fargate situation as EC2 approach will not have task definition.

I can't add production DB details in docker run command, As I am storing them in AWS Secretmanager. Let me know what is the best approach to add db details to wordpress docker container.
